matryoshka_tester/trigger_test.py

```diff
@@ -0,0 +1,119 @@
+import json
+from os import environ
+from typing import Dict, Literal, List, TypedDict, Union
+
+import pika
+import requests
+
+from matryoshka_tester.parse_data import containers
+
+
+class PublishBodyBase(TypedDict):
+    #: name of the project which' publishing state changed
+    project: str
+    #: repository which' publishing state changed
+    repo: str
+
+
+class PacktrackBody(PublishBodyBase):
+    #: some id identifying something
+    payload: str
+
+
+class PublishStateBody(PublishBodyBase):
+    #: the new state of the project + repository
+    state: Literal[
+        "unknown",
+        "broken",
+        "scheduling",
+        "blocked",
+        "building",
+        "finished",
+        "publishing",
+        "published",
+        "unpublished",
+    ]
+
+
+class PublishedBody(PublishBodyBase):
+    #: some internal (=undocumented) build ID
+    buildid: str
+
+
+connection = pika.BlockingConnection(
+    pika.URLParameters("amqps://opensuse:opensuse@rabbit.opensuse.org")
+)
+channel = connection.channel()
+
+exchange = channel.exchange_declare(
+    exchange="pubsub", exchange_type="topic", passive=True, durable=True
+)
+
+result = channel.queue_declare("", exclusive=True)
+queue_name = result.method.queue
+
+channel.queue_bind(exchange="pubsub", queue=queue_name, routing_key="#")
+
+
+PREFIX = "opensuse.obs"
+REPO_ROUTING_KEYS = [
+    # f"{PREFIX}.repo.packtrack",
+    f"{PREFIX}.repo.publish_state",
+    f"{PREFIX}.repo.published",
+]
+
+
+watched_projects: Dict[str, List[str]] = {}
```

Comment:
same here, I would just put this into a function for unit testing.
This means the watched projects need to be passed around, in the 'main' ('trigger_from_mqtt') function.

Reducing the global variables will things also explicit for callback.

Reply:
Done, this is now in the class attribute `RabbitMQConnection.watched_projects`